src/orion/core/worker/producer.py

```diff
@@ -24,32 +25,58 @@ class Producer(object):
 
     """
 
-    def __init__(self, experiment):
+    def __init__(self, experiment, max_attemps=100):
         """Initialize a producer.
 
         :param experiment: Manager of this experiment, provides convenient
            interface for interacting with the database.
         """
         log.debug("Creating Producer object.")
         self.experiment = experiment
-        self.num_new_trials = experiment.pool_size
         self.space = experiment.space
         if self.space is None:
             raise RuntimeError("Experiment object provided to Producer has not yet completed"
                                " initialization.")
         self.algorithm = experiment.algorithms
+        self.max_attemps = max_attemps
+
+    @property
+    def pool_size(self):
+        """Pool-size of the experiment"""
+        return self.experiment.pool_size
 
     def produce(self):
         """Create and register new trials."""
-        log.debug("### Suggest new ones.")
-        new_points = self.algorithm.suggest(self.num_new_trials)
-
-        log.debug("### Convert them to `Trial` objects.")
-        new_trials = list(map(lambda data: format_trials.tuple_to_trial(data, self.space),
-                              new_points))
-
-        log.debug("### Register to database: %s", new_trials)
-        self.experiment.register_trials(new_trials)
+        sampled_points = 0
+        n_attemps = 0
+
+        while sampled_points < self.pool_size and n_attemps < self.max_attemps:
+            n_attemps += 1
+            log.debug("### Suggest new ones.")
```

Comment:
I feel like this could be changed to something more explicit like `Suggesting new trials` since it meant for debug purposes.

Reply:
And here's another reason why : I thought this was suggesting trials, but it points.

Reply:
Agreed!